lib/SparqlGenerator.js

```diff
@@ -31,11 +31,12 @@ Generator.prototype.toQuery = function (q) {
   if (q.distinct)
     query += 'DISTINCT ';
 
-  if (q.variables)
+  if (q.variables){
     query += mapJoin(q.variables, undefined, function (variable) {
-      return isString(variable) ? this.toEntity(variable) :
-             '(' + this.toExpression(variable.expression) + ' AS ' + variable.variable + ')';
+      return isTerm(variable) ? this.toEntity(variable) :
+             '(' + this.toExpression(variable.expression) + ' AS ' + "?" + variable.variable.value + ')';
```

Comment:
I'm surprised the linter doesn't complain about this, but the convention in this project is to write strings with single quotes. (double quotes are used in several places throughout this PR)

Reply:
Also, the `'?' + variable.variable.value` pattern occurs several times. I'd suggest to abstract this into a `variableToString` method.